### ISSUE TITLE
ES6 Object key shorthand not compiling to ES5

### ISSUE THREAD

**Issue by evenicoulddoit** [google/closure-compiler#1121]
Traceback:

```
[09:06:06] Command failed: java -jar -XX:+TieredCompilation node_modules/google-closure-compiler/compiler.jar --flagfile="/tmp/e87c80e4-8aa3-49fb-bcfb-d748cc4621e4" --compilation_level=WHITESPACE_ONLY --create_source_map=project/angular/dist/app.min.js.map --source_map_location_mapping=project/angular/|../ --language_in=ECMASCRIPT6_STRICT --language_out=ECMASCRIPT5_STRICT --jscomp_off=misplacedTypeAnnotation --output_wrapper=%output%
//# sourceMappingURL=app.min.js.map --js_output_file="app.min.js"
java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

null
  Node(ASSIGN): project/angular/components/app/appController.js:28:6
      this.alert = {msg, type: 'danger'};
  Parent(EXPR_RESULT): project/angular/components/app/appController.js:28:6
      this.alert = {msg, type: 'danger'};

  at com.google.javascript.jscomp.NodeUtil.checkForStateChangeHelper(NodeUtil.java:923)
  at com.google.javascript.jscomp.NodeUtil.checkForStateChangeHelper(NodeUtil.java:935)
  at com.google.javascript.jscomp.NodeUtil.checkForStateChangeHelper(NodeUtil.java:1025)
  at com.google.javascript.jscomp.NodeUtil.mayHaveSideEffects(NodeUtil.java:910)
  at com.google.javascript.jscomp.CheckSideEffects.visit(CheckSideEffects.java:121)
  at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:594)
  at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:588)
  at com.google.javascript.jscomp.NodeTraversal.traverseBlockScope(NodeTraversal.java:637)
  at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:582)
  at com.google.javascript.jscomp.NodeTraversal.traverseFunction(NodeTraversal.java:628)
  at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:580)
  at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:588)
  at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:588)
  at com.google.javascript.jscomp.NodeTraversal.traverseBlockScope(NodeTraversal.java:637)
  at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:582)
  at com.google.javascript.jscomp.NodeTraversal.traverseFunction(NodeTraversal.java:628)
  at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:580)
  at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:588)
  at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:588)
  at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:588)
  at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:588)
  at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:588)
  at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:588)
  at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:588)
  at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:588)
  at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:588)
  at com.google.javascript.jscomp.NodeTraversal.traverseBlockScope(NodeTraversal.java:637)
  at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:582)
  at com.google.javascript.jscomp.NodeTraversal.traverseFunction(NodeTraversal.java:628)
  at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:580)
  at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:588)
  at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:588)
  at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:588)
  at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:588)
  at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:295)
  at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:545)
  at com.google.javascript.jscomp.CheckSideEffects.process(CheckSideEffects.java:72)
  at com.google.javascript.jscomp.PhaseOptimizer$NamedPass.process(PhaseOptimizer.java:285)
  at com.google.javascript.jscomp.PhaseOptimizer.process(PhaseOptimizer.java:217)
  at com.google.javascript.jscomp.Compiler.check(Compiler.java:841)
  at com.google.javascript.jscomp.Compiler.compileInternal(Compiler.java:757)
  at com.google.javascript.jscomp.Compiler.access$000(Compiler.java:95)
  at com.google.javascript.jscomp.Compiler$3.call(Compiler.java:656)
  at com.google.javascript.jscomp.Compiler$3.call(Compiler.java:653)
  at com.google.javascript.jscomp.Compiler$4.call(Compiler.java:700)
  at java.util.concurrent.FutureTask.run(FutureTask.java:262)
  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
  at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NullPointerException
  ... 49 more
```

**Comment by MatrixFrog** [comment]
https://closure-compiler-debugger.appspot.com/#input0%3Dvar%2520msg%2520%253D%2520'oh%2520no!'%253B%250Athis.alert%2520%253D%2520%257Bmsg%252C%2520type%253A%2520'danger'%257D%253B%26input1%26conformanceConfig%26externs%26refasterjs-template%26includeDefaultExterns%3D1%26LANG_IN_IS_ES6%3D1%26PRETTY_PRINT%3D1%26TRANSPILE%3D1

**Comment by MatrixFrog** [comment]
But you can make it not happen by changing the left hand side (!?)

https://closure-compiler-debugger.appspot.com/#input0%3Dvar%2520msg%2520%253D%2520'oh%2520no!'%253B%250Avar%2520alert%2520%253D%2520%257Bmsg%252C%2520type%253A%2520'danger'%257D%253B%26input1%26conformanceConfig%26externs%26refasterjs-template%26includeDefaultExterns%3D1%26LANG_IN_IS_ES6%3D1%26PRETTY_PRINT%3D1%26TRANSPILE%3D1

**Comment by MatrixFrog** [comment]
The transpilation actually works fine. Or rather it *would* work fine, but it never runs because the CheckSideEffects pass now comes before transpilation.

**Comment by Dominator008** [comment]
Seems like NodeUtil is missing something?

**Comment by MatrixFrog** [comment]
Yup. https://github.com/google/closure-compiler/blob/cf7c26765722bbd37e7569f551ab0dbdbb18d42d/src/com/google/javascript/jscomp/NodeUtil.java#L945

Fix in progress.

**Comment by MatrixFrog** [comment]
Fixed in 2f4b4d383b817d20677dad7f55739675ce7ba022
